internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -59,7 +59,7 @@ func New(cmd Cmd) CmdClient {
 // BuildArguments holds the arguments that can be passed while building a container.
 type BuildArguments struct {
 	URI        string            // Required. Location of ECR Repo. Used to generate image name in conjunction with tag.
-	Tags       []string          // Optional. List of tags to apply to the image besides "latest".
```

Comment:
Why did it change from optional to required? We still use git short commit id if user doesn't provide tag right?

Reply:
Previously if the client does not provide any `Tags` then docker build defaults to `latest`.  But now the client should explicitly provide  `latest` tag or any custom tags for building and pushing image. This makes the `Tags` Required if that makes sense.

Reply:
If client doesn't provide tag then what happens? Do we error out in that situation?

Reply:
If the tags are empty the current behavior is building all the images with default `latest` tag and not pushing any images to the ECR repository which is weird.

Good call to call 👍  to error out if the tags are empty so that we do not even build and push images if the tags are empty.